### PR TITLE
Added example showing using multiple source signals with createResource

### PR DESCRIPTION
**Pull request by Tommypop2**
Added example showing using multiple source signals with createResource, and passing those values into the fetcher

### CODE REVIEW

File: content/references/api-reference/basic-reactivity/createResource.mdx [hunk 1]
```diff
@@ -99,6 +99,19 @@ mutate(optimisticValue);
...
 
+It's also possible to pass multiple signals as the source, by constructing a derived signal:
+
+```ts
+const [dataSignal, setDataSignal] = createSignal();
+const [moreData, setMoreData] = createSignal();
+const [data] = createResource(
+  () => [dataSignal(), moreData()],
+  ([dataVal, moreDataVal]) => {
+    // Reruns when either signal updates
```
Comment (michaelessiet) [inline content/references/api-reference/basic-reactivity/createResource.mdx:110]: Thank you for adding this. Please can you add a note that when using this in typescript the values need to be explicitly typed e.g 
Screenshot 2023-07-15 at 15 01 47

**Review by michaelessiet — CHANGES_REQUESTED** [review]

File: content/references/api-reference/basic-reactivity/createResource.mdx (hunk 1 above)
Comment (Tommypop2) [reply content/references/api-reference/basic-reactivity/createResource.mdx:110]: Thanks for the review.
It's also possible to type this properly through `as const`: https://playground.solidjs.com/anonymous/c8210975-e41f-46ce-9488-583d1b54b686
Would it be better to use `as const` in the example instead, and state that typescript requires explicit typing?

Comment (Tommypop2) [reply content/references/api-reference/basic-reactivity/createResource.mdx:110]: I've just updated it with `as const`, and why it should be included, as well as the type-assertions method as an alternative

Comment (michaelessiet) [reply content/references/api-reference/basic-reactivity/createResource.mdx:110]: Great work! I'll merge this tomorrow

**Review by michaelessiet — APPROVED** [review]

**Pull request merged**